### PR TITLE
Frontend user profile

### LINKED ISSUE #32: Users should be able to get a user detail

### LINKED ISSUE #79: The "Last Solve" column in ranking shows an incorrect time

### PR DESCRIPTION
**Pull request by EricLung**
Again, not focus on the UI design but the functions

Closes #32, closes #79.

### CODE REVIEW

File: front-end/src/store/users.js [hunk 1]
```diff
@@ -64,6 +71,14 @@ export default {
       Vue.set(state.listRank, 'errors', [])
       Vue.set(state.listRank, 'errorAlert', false)
     },
+    resetGetUser (state) {
+      Vue.set(state.getUser, 'user', {})
+      Vue.set(state.getUser, 'solvedChallenges', {})
```
Comment (samueltangz) [inline front-end/src/store/users.js]: `challengeSolved` or `solvedChallenges`?

File: front-end/src/views/User.vue [hunk 2]
```diff
@@ -7,41 +7,135 @@
...
+      </b-col>
+    </b-row>
+    <b-row align-h="center">
+      Solved Challenges
+    </b-row>
+    <b-row>
+      <b-col>
+        <b-table
+          :fields="solvedChallengeTableFields"
+          :items="getUser.user.challengesSolved"
```
Comment (samueltangz) [inline front-end/src/views/User.vue]: `challengeSolved` or `solvedChallenges`?

File: front-end/src/views/User.vue [hunk 3]
```diff
@@ -7,41 +7,135 @@
     </b-row>
     <b-row>
       <b-col>
-        {{ user }}
+        <b-alert
```
Comment (samueltangz) [inline front-end/src/views/User.vue]: why do we need to show the error twice?
```
    <b-row>
      <b-col>
        <b-alert
          v-model="getUser.errorAlert"
          variant="danger"
          dismissible>
          {{ getUser.error }}
        </b-alert>
      </b-col>
    </b-row>
```

File: front-end/src/views/User.vue [hunk 4]
```diff
@@ -7,41 +7,135 @@
...
       </b-col>
     </b-row>
   </b-container>
 </template>
 
 <script>
+import { mapState } from 'vuex'
+
+import store from '@/store'
+import time from '@/utils/time'
```
Comment (samueltangz) [inline front-end/src/views/User.vue]: one extra line between `import` and `export`

File: front-end/src/store/users.js [hunk 5]
```diff
@@ -159,6 +174,25 @@ export default {
...
       }
+    },
+    async getUser ({ state, commit }, id) {
+      commit('resetGetUser')
+      state.getUser.loading = true
+      try {
+        const res = await axios.get(`/api/users/${id}`)
+        const { user } = res.data
+        state.getUser.user = user
+        // test
```
Comment (samueltangz) [inline front-end/src/store/users.js]: use `// TODO` instead... we can find this line easier later

File: front-end/src/store/users.js [hunk 6]
```diff
@@ -64,6 +71,14 @@ export default {
       Vue.set(state.listRank, 'errors', [])
       Vue.set(state.listRank, 'errorAlert', false)
     },
+    resetGetUser (state) {
+      Vue.set(state.getUser, 'user', {})
+      Vue.set(state.getUser, 'solvedChallenges', {})
+      Vue.set(state.getUser, 'useradiuser', false)
```
Comment (samueltangz) [inline front-end/src/store/users.js]: ?????

File: front-end/src/store/users.js (hunk 1 above)
Comment (samueltangz) [reply front-end/src/store/users.js]: (should not exist anyway...)

Comment (EricLung) [reply front-end/src/store/users.js]: forgot to delete -_-

File: front-end/src/store/users.js (hunk 6 above)
Comment (samueltangz) [reply front-end/src/store/users.js]: btw there should be a line for `loading`

File: front-end/src/views/User.vue (hunk 2 above)
Comment (EricLung) [reply front-end/src/views/User.vue]: `challengeSolved` according to backend respond

**Pull request merged**